ffcx/ir/integral.py

```diff
@@ -80,11 +80,17 @@ def compute_integral_ir(cell, integral_type, entitytype, integrands, argument_sh
         # terminal_data again after factorization if that's necessary.
 
         initial_terminals = {
-            i: analyse_modified_terminal(v["expression"])
+            i: analyse_modified_terminal(v["expression"], cell)
             for i, v in S.nodes.items()
             if is_modified_terminal(v["expression"])
         }
 
+        # Check if we have a mixed-dimensional integral
+        is_mixed_dim = False
+        for domain in integrand.ufl_domains():
+            if domain.topological_dimension() != cell.topological_dimension():
+                is_mixed_dim = True
+
```

Comment:
Is there a reason to use a boolean `(is_mixed_dim)` instead of computing `codim = cell.topological_dimension() - domain.topological_dimension()` ?

Reply:
`is_mixed_dim` indicates if an integral contains any mixed-dimensional terms. In `build_optimized_tables`, we need to differentiate between the codim 0 element and the codim 1 element in a mixed-dimensional integral so that only the quadrature rule for the codim 0 element is permuted (see [here](https://github.com/FEniCS/ffcx/pull/675/files#diff-b2e2e37c190c9887e10625dad288f14dfc31c596ffab0ca77602d66399c0a019R367)). I've added a check that `codim > 0` when the codimension of the modified terminal is computed [here](https://github.com/FEniCS/ffcx/pull/675/files#diff-b2e2e37c190c9887e10625dad288f14dfc31c596ffab0ca77602d66399c0a019R358). An alternative approach might be to replace `is_mixed_dim` with something like `integral_codim` and rename `codim` to `element_codim`. Let me know which you prefer